org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/internal/core/manipulation/JavaManipulationPlugin.java

```diff
@@ -43,6 +43,8 @@ public class JavaManipulationPlugin extends Plugin implements DebugOptionsListen
 
 	public static boolean DEBUG_AST_PROVIDER;
 
+	public static boolean DEBUG_TYPE_CONSTRAINTS;
```

Comment:
Would it make sense to remove https://github.com/eclipse-jdt/eclipse.jdt.ui/blob/939c9911577144372fff1d15b8211c49012ff221/org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/JavaPlugin.java#L166 ? You've migrated all the points in the code that would have used that boolean.